crates/ink/ir/src/ir/item_impl/mod.rs

```diff
@@ -373,3 +379,35 @@ impl ItemImpl {
         &self.items
     }
 }
+
+/// Check if the the conditional compilation expression
+/// for any `cfg` attributes is satisfied.
+///
+/// # Example
+/// If we have an annotated item (e.g. message)
+/// ```ignore
+/// #[ink(message)]
+/// #[cfg(feature = "foo")]
+/// #[cfg(feature = "baz")]
+/// pub fn get(&self) -> String {
+///     self.value.clone()
+/// }
+/// ```
+/// The function would would iterate over the `cfg` attributes
+/// and return `true` if any of the feature flags are set
+pub fn is_code_span_enabled(attrs: &[Attribute]) -> bool {
+    let cfg_attrs = attrs
+        .iter()
+        .filter(|a| a.path.is_ident("cfg"))
+        .map(|a| &a.tokens);
+    if cfg_attrs.clone().count() == 0 {
+        return true
+    }
+    for attr_tokens in cfg_attrs {
+        let _s = attr_tokens.to_token_stream();
+        if cfg!(_s) {
+            return true
```

Comment:
I believe this logic is wrong in the case of multiple `cfg` attributes e.g.
```
fn main() {
    foo_bar();
}

#[cfg(feature = "foo")]
#[cfg(feature = "bar")]
fn foo_bar() {}
```
`cargo build --features foo` -> :red_circle: 
`cargo build --features bar` -> :red_circle: 
`cargo build --features foo,bar` -> :green_circle: 

So having separate cfg attrs seems to be the equivalent of `#[cfg(all(feature = "foo", feature = "bar"))]` 

The implementation here is instead `#[cfg(any(feature = "foo", feature = "bar"))]`.

Reply:
The logic would be easier to read when expressed e.g. in the form:

```rust
pub fn is_code_span_enabled(attrs: &[Attribute]) -> bool {
    attrs
        .iter()
        .filter(|a| a.path.is_ident("cfg"))
        .all(|a| {
            let predicate = &a.tokens.to_token_stream();
            cfg!(predicate)
        })
}
```